app/sprinkles/account/src/Controller/AccountController.php

```diff
@@ -500,52 +428,42 @@ public function login(Request $request, Response $response, $args)
+            if ($result['breaches'] > $this->ci->config['site.password_security.enforce_no_compromised']) {
 
-            $password = $data['password'];
+              // Try to generate a new password reset request.
+                // Use timeout for "reset password"
+                $passwordReset = $this->ci->repoPasswordReset->create($currentUser, $config['password_reset.timeouts.reset']);
+                $token = ['token'=> $passwordReset->getToken()];
+
+                // Load validation rules - note this uses the same schema as "set password"
+                $schema = new RequestSchema('schema://requests/set-password.yaml');
+                $validator = new JqueryValidationAdapter($schema, $this->ci->translator);
+
+                $target = $this->ci->router->pathFor('set-password', [
+                'page' => [
+                  'validators' => [
+                      'set_password'    => $validator->rules('json', false)
+                  ]
+                ]
+              ], $token);
 
-            $result = $this->checkPassword($password);
+                $ms->addMessageTranslated('info', 'PASSWORD.SECURITY.RESET_REQUIRED');
 
-            if ($result['breaches'] > $this->ci->config['site.password_security.enforce_no_compromised']) {
-                $allowedPassword = false;
+                return $response->withRedirect($target);
             }
         }
 
-        if ($allowedPassword == false) {
-
-            // Try to generate a new password reset request.
-            // Use timeout for "reset password"
-            $passwordReset = $this->ci->repoPasswordReset->create($currentUser, $config['password_reset.timeouts.reset']);
-
-            // Load validation rules - note this uses the same schema as "set password"
-            $schema = new RequestSchema('schema://requests/set-password.yaml');
-            $validator = new JqueryValidationAdapter($schema, $this->ci->translator);
+        $ms->addMessageTranslated('success', 'WELCOME', $currentUser->export());
 
-            $token = ['token'=> $passwordReset->getToken()];
+        // Set redirect, if relevant
+        $redirectOnLogin = $this->ci->get('redirect.onLogin');
 
-            $target = $this->ci->router->pathFor('set-password', [
-              'page' => [
-                'validators' => [
-                    'set_password'    => $validator->rules('json', false)
-                ]
-            ]
-          ], $token);
-
-            $ms->addMessageTranslated('info', 'PASSWORD.SECURITY.RESET_REQUIRED');
-
-            return $response->withRedirect($target);
-            exit;
-        } else {
-            $ms->addMessageTranslated('success', 'WELCOME', $currentUser->export());
-
-            // Set redirect, if relevant
-            $redirectOnLogin = $this->ci->get('redirect.onLogin');
-
-            return $redirectOnLogin($request, $response, $args);
-        }
+        return $redirectOnLogin($request, $response, $args);
```

Comment:
Any ideas on why this would `return` twice? @lcharette said I need an exorcist. 
This line is still getting returned after line 457 above is sent. (Sometimes it works correctly when I test it) See screenshot below. 

![image](https://user-images.githubusercontent.com/33728190/57453950-ac3ca700-7235-11e9-9610-1e5000441d2c.png)

Reply:
Not sure I follow. It should be impossible for a `return` statement to be triggered twice unless the client makes 2 requests. Some reproduction steps plus expected and actual outcomes might help.